lib/math/math.go

```diff
@@ -162,6 +162,37 @@ func log(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwar
 	return starlark.Float(math.Log(float64(x)) / math.Log(float64(base))), nil
 }
 
+// ceil wraps the Ceil function as a Starlark built-in that
+// returns the ceiling as an Integral.
+func ceil(
+	thread *starlark.Thread,
+	_ *starlark.Builtin,
+	args starlark.Tuple,
+	kwargs []starlark.Tuple,
+) (starlark.Value, error) {
+	var x floatOrInt
```

Comment:
It is not correct to use floatOrInt here. If x is an int, we should return it unchanged, not convert it to a float, which may lose information.

Reply:
Great point, thanks for pointing that out.